libcloud/compute/drivers/azure.py

```diff
@@ -985,6 +984,70 @@ def ex_set_instance_endpoints(self, node, endpoints,
 
         self.raise_for_response(response, 202)
 
+    def ex_create_storage_service(self, name, location,
+                                  description=None, affinity_group=None,
+                                  extended_properties=None):
+        """
+        Create an azure storage service.
+
+        :param      name: Name of the service to create
+        :type       name: ``str``
+
+        :param      location: Standard azure location string
+        :type       location: ``str``
+
+        :param      description: (Optional) Description of storage service.
+        :type       description: ``str``
+
+        :param      affinity_group: (Optional) Azure affinity group.
+        :type       affinity_group: ``str``
+
+        :param      extended_properties: (Optional) Additional configuration
+                                         options support by Azure.
+        :type       extended_properties: ``dict``
+
+        :rtype: ``bool``
+        """
+
+        response = self._perform_storage_service_create(
+            self._get_storage_service_path(),
+            AzureXmlSerializer.create_storage_service_to_xml(
+                service_name=name,
+                label=self._encode_base64(name),
```

Comment:
Ignore my comment. I've confirmed it does indeed need to be base64 encoded - https://msdn.microsoft.com/en-us/library/azure/hh264518.aspx